Documentation/api.md

```diff
@@ -395,6 +395,7 @@ PrometheusSpec is a specification of the desired behavior of the Prometheus clus
 | portName | Port name used for the pods and governing service. This defaults to web | string | false |
 | arbitraryFSAccessThroughSMs | ArbitraryFSAccessThroughSMs configures whether configuration based on a service monitor can access arbitrary files on the file system of the Prometheus container e.g. bearer token files. | [ArbitraryFSAccessThroughSMsConfig](#arbitraryfsaccessthroughsmsconfig) | false |
 | overrideHonorLabels | OverrideHonorLabels if set to true overrides all user configured honor_labels. If HonorLabels is set in ServiceMonitor or PodMonitor to true, this overrides honor_labels to false. | bool | false |
+| enforcedNamespaceLabel | EnforcedNamespaceLabel adds a namespace label of origin for each alert and metric that is user created. | string | false |
```

Comment:
I had no clue how to phrase this nicely and descriptive enough, suggestions welcome! :)

Reply:
I think I like this name.

Reply:
Not name, I mean the description suggestions of what this does :)

Reply:
maybe instead of "adds" we could say explicitely say "enforces"? The only question mark I had was "of origin". Maybe we could also summarize the edge cases in the description too (as is done in the PR title)?

Reply:
Does this sgty @s-urbaniak @brancz 
```
EnforcedNamespaceLabel enforces adding a namespace label of origin for each alert
and metric that is user created. The label value will always be the namespace of the object that is
being created.
```

We have very short descriptions, I think the rest belongs in the docs IMO. I can do that in a follow up, if thats okay with you.

Reply:
sounds good! :+1: